adal/authority.py

```diff
@@ -60,7 +60,8 @@ def __init__(self, authority_url, validate_authority=True):
 
     @property
     def url(self):
-        return self._url.geturl()
+        return "https://{}/{}".format(self._host,

```

Comment:
Hmm, this technically changes the content of `authority.url`. Before this PR, it was basically the same url being passed in to the Authority constructor. After this PR, it is truncated to the tenant part. This may still be OK, but we will need to examine our entire code base to find all its usages, before we can make sure it won't break anything. Did we check that?

Alternatively, instead of silently changing the raw input and store it here, we could leave it as-is so that we maintain the previous implicit contract: "`self.url` is basically the same as raw input"; that way you don't even need to modify (and duplicate) the logic at [line 158](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/151/files#diff-03f44f824a1959ef23cc5f4861b150a5R158) and [line 161](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/151/files#diff-03f44f824a1959ef23cc5f4861b150a5R161). "But then how do we detect and deal with misformed input?" I would suggest we raise ValueError exception. This is the philosophy of [fail early, fail loudly](http://oncodingstyle.blogspot.com/2008/10/fail-early-fail-loudly.html).